File: tournaments/pagesNumbering/pagesNumbering.cpp
int pagesNumbering(int n) {

  int tenPower = 1,
      digitsPerPage = 1,
      result = 0;

  while (tenPower * 10 <= n) {
    result += tenPower * 9 * digitsPerPage;
    tenPower *= 10;
    digitsPerPage++;
  }
  result += (n - tenPower + 1) * digitsPerPage;

  return result;
}

File: tournaments/pagesNumbering/pagesNumbering.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub pagesNumbering {
    my ($n) = @_;

    my $result = 0;
    my $x = 1;
    my $y = 1;

    while ($n >= 10 * $x) {
        $result += 9 * $x * $y;
        $x *= 10;
        ++$y;
    }

    $result += ($n - $x + 1) * $y;

    return $result;
}
